inference-engine/src/transformations/src/transformations/op_conversions/simplify_ctc_greedy_decoder_seq_len.cpp

```diff
@@ -0,0 +1,111 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "itt.hpp"
+
+#include "transformations/op_conversions/simplify_ctc_greedy_decoder_seq_len.hpp"
+
+#include <ngraph/opsets/opset6.hpp>
+#include <ngraph/rt_info.hpp>
+
+#include <ngraph/pattern/op/wrap_type.hpp>
+
+NGRAPH_RTTI_DEFINITION(ngraph::pass::SimplifyCTCGreedyDecoderSeqLen, "SimplifyCTCGreedyDecoder", 0);
+
+ngraph::pass::SimplifyCTCGreedyDecoderSeqLen::SimplifyCTCGreedyDecoderSeqLen() {
+    MATCHER_SCOPE(SimplifyCTCGreedyDecoderSeqLen);
+    auto decoder = pattern::wrap_type<opset6::CTCGreedyDecoderSeqLen>();
+
+    ngraph::matcher_pass_callback callback = [=](ngraph::pattern::Matcher& m) {
+        auto decoder_seq_len = std::dynamic_pointer_cast<opset6::CTCGreedyDecoderSeqLen> (m.get_match_root());
+        if (!decoder_seq_len) {
+            return false;
+        }
```

Comment:
Why do we need this check?
The transformation should not work if `blank_index != C-1`. And such check is needed.

Reply:
I believe this is just check for nullptr..

Reply:
We cannot check blank_index and C. They are part of graph.

Reply:
Why can't we check? If it is a constant, we can retrieve a value.